public/app/plugins/datasource/graphite/graphite_query.ts

```diff
@@ -326,3 +330,44 @@ export default class GraphiteQuery {
 function renderTagString(tag: { key: any; operator?: any; value?: any }) {
   return tag.key + tag.operator + tag.value;
 }
+
+/**
+ * mutates the second seriesByTag function into a string to fix a parsing bug
+ * @param astNode
+ * @param innerFunc
+ */
+// eslint-disable-next-line
+function handlemultipleSeriesByTagsParams(astNode: any, innerFunc: any) {
+  // if function is asPercent and has two params that are function seriesByTags keep the second as a string otherwise we have a parsing error
+  if (innerFunc.def.name === 'asPercent' && astNode.params.length >= 2) {
+    let count = 0;
+    // eslint-disable-next-line
+    astNode.params = astNode.params.map((p: any) => {
+      if (p.type === 'function') {
+        count += 1;
+      }
+
+      if (count === 2 && p.type === 'function' && p.name === 'seriesByTag') {
+        // convert second function to a string
+        // eslint-disable-next-line
+        const stringParams = p.params.reduce(
+          (acc: string, p: { type: string; value: string }, idx: number, paramsArr: any[]) => {
+            if (idx === 0 || idx !== paramsArr.length - 1) {
+              return `${acc}'${p.value}',`;
+            }
+
+            return `${acc}'${p.value}'`;
+          },
+          ''
+        );
+
+        return {
+          type: 'string',
+          value: `${p.name}(${stringParams})`,
+        };
+      }
+
+      return p;
+    });
+  }
+}
```

Comment:
Here's some types to avoid eslint-ignores
```suggestion

type ASTNodeParams = {
  type: string
  name?: string
  value: string
  params?: ASTNodeParams[]
}

interface ASTNode {
  params: ASTNodeParams[]
}

/**
 * mutates the second seriesByTag function into a string to fix a parsing bug
 * @param astNode
 * @param innerFunc
 */
function handlemultipleSeriesByTagsParams(astNode: ASTNode, innerFunc: any) {
  // if function is asPercent and has two params that are function seriesByTags keep the second as a string otherwise we have a parsing error
  if (innerFunc.def.name === 'asPercent' && astNode.params.length >= 2) {
    let count = 0;
    astNode.params = astNode.params.map((p: ASTNodeParams) => {
      if (p.type === 'function') {
        count += 1;
      }

      if (count === 2 && p.type === 'function' && p.name === 'seriesByTag') {

        // convert second function to a string
        const stringParams = p.params?.reduce(
          (acc: string, p: ASTNodeParams, idx: number, paramsArr: any[]) => {
            if (idx === 0 || idx !== paramsArr.length - 1) {
              return `${acc}'${p.value}',`;
            }

            return `${acc}'${p.value}'`;
          },
          ''
        );

        return {
          type: 'string',
          value: `${p.name}(${stringParams})`,
        };
      }

      return p;
    });
  }
}
```

Reply:
kinda rough on formatting, and the types should go to the top of the file, but lemme know what you think

Reply:
even if these definitions aren't entirely complete, if we use these more and more over time, we'll update the definitions and they should get closer to being correct

Reply:
Cool! @gtk-grafana I have this PR up for the types https://github.com/grafana/grafana/pull/67067 which adds a little more to the ast node definition. Can we use this? We'll need even more to make it applicable to the other areas where ast nodes are applied to parts of the query, but it's a work in progress, a little at a time.

Reply:
Yes please, let's merge that on in! I'd like to avoid adding any `any` or eslint ignores, even if we need to do partial refactors like this or just define types for new code as I suggested.

Reply:
I think this is a good goal! to avoid using `any` or eslint ignores is the optimum. I also want to mention, there is a price to refactoring legacy code that is littered with `any`. This took me a day to refactor and broke almost every test in the parser. I agree we should try to implement a solution like yours but also be aware that it can require a significant time investment. When it comes to investing time, I think we should bring issues like these up in quarterly planning.